.github/workflows/typescript-cli-checks.yml

```diff
@@ -0,0 +1,19 @@
+name: typescript-cli-checks
+on:
+  push:
+    paths: [
+      '.github/workflows/**/*',
+      'cli/**/*',
+      'typescript-cli/**/*'
+    ]
+jobs:
+  typescript-cli:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '18'
+      - run: npm install --save-dev
```

Comment:
You are probably better using `npm ci` here. It uses the lock file for resolving packages, which is faster & more reproducible.

Also `--save-dev` saves packages to `devDependencies`. All dev dependencies are installed by default. They only won't be installed if you use `--omit=dev`.

Reply:
Fixed, thanks.